File: src/BaseHmiTypes/Screens/Base/HmiSliderAnimation.cs
namespace BaseHmiTypes.Screens.Base;

/// <summary>
/// Describes a writable slider animation that maps object movement to a tag value.
/// </summary>
public class HmiSliderAnimation
{
    public string? Tag { get; set; }

    public HmiAnimationRangeSource? RangeSource { get; set; }

    public double? TagMinimum { get; set; }

    public double? TagMaximum { get; set; }

    public string? MinimumTag { get; set; }

    public string? MaximumTag { get; set; }

    public double? OffsetMinimum { get; set; }

    public double? OffsetMaximum { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiSliderAnimation.ts
import { HmiAnimationRangeSource } from "./HmiAnimationRangeSource.js";

/** Describes a writable slider animation that maps object movement to a tag value. */
export class HmiSliderAnimation {
  tag?: string;
  rangeSource?: HmiAnimationRangeSource;
  tagMinimum?: number;
  tagMaximum?: number;
  minimumTag?: string;
  maximumTag?: string;
  offsetMinimum?: number;
  offsetMaximum?: number;
}
